benchmarks/utils.nim

```diff
@@ -0,0 +1,54 @@
+template withDir*(dir: string, blk: untyped) =
+  ## set working dir for duration of blk
+  let prev = getCurrentDir()
+  try:
+    setCurrentDir(dir)
+    `blk`
+  finally:
+    setCurrentDir(prev)
+
+template runit*(cmd: string) =
+  ## run shell commands and verify it runs without an error code
+  echo "RUNNING: ", cmd
+  let cmdRes = execShellCmd(cmd)
+  echo "STATUS: ", cmdRes
+  assert cmdRes == 0
+
+template benchmark*(benchmarkName: string, blk: untyped) =
+  ## simple benchmarking of a block of code
+  let nn = 5
+  var vals = newSeqOfCap[float](nn)
+  for i in 1 .. nn:
+    block:
+      let t0 = epochTime()
+      `blk`
+      let elapsed = epochTime() - t0
+      vals.add elapsed
+
+  var elapsedStr = ""
+  for v in vals:
+    elapsedStr &= ", " & v.formatFloat(format = ffDecimal, precision = 3)
+  stdout.styledWriteLine(
+    fgGreen, "CPU Time [", benchmarkName, "] ", "avg(", $nn, "): ", elapsedStr, " s"
```

Comment:
Can we actually derive an average for the number of runs here? in fact, I would print this on it's own line at the end of the bench run.

Something like:

```sh
"CPU Time ["... "], ...s"
...

"Average for ${nn} runs is ...."
```

Reply:
Definitely, I was thinking about doing that. The current form is because I wanted to print out each run so I could do a simple linear regression on it. It's a pain to work with though in this form.